elastic/kibana
# [console] Deprecate "proxyFilter" and "proxyConfig" on 8.x

Remove support for the "proxyFilter" and "proxyConfig" Kibana setting in console from `8.x`.

## How to test

Add the following in your `kibana.yml`

```
console.proxyFilter:
  - ^https?://(localhost|127\.0\.0\.1|\[::0\]).*
console.proxyConfig:
  - match:
      host: "*.internal.org" # allow any host that ends in .internal.org
      port: "{9200..9299}" # allow any port from 9200-9299
      protocol: "foo"
      path: "bar"
    timeout: 1000
    ssl:
      ca:
        - "/opt/certs/internal.ca"
      verify: false
      cert: "foo"
      key: "bar"
```

The Kibana server should throw an error mentioning that those settings are unknown.

Now change the `MAJOR_VERSION` constant in `src/plugins/console/common/constants/plugin.ts` to `7.16.0`. The server will restart and won't throw an error for those settings anymore.

### Test the deprecation in Upgrade Assistant:

* Open `x-pack/plugins/upgrade_assistant/public/plugin.ts`
* [Change L20](https://github.com/elastic/kibana/blob/master/x-pack/plugins/upgrade_assistant/public/plugin.ts#L20) with `const readonly = false;`
* Navigate to Upgrade Assistant. You should see the 2 console settings under critical

<img width="1254" alt="Screenshot 2021-10-01 at 19 41 32" src="https://user-images.githubusercontent.com/2854616/135672290-1ee52f1a-414d-4373-8bb0-9a9bbac379b2.png">

<img width="873" alt="Screenshot 2021-10-01 at 19 41 43" src="https://user-images.githubusercontent.com/2854616/135672310-15e7347d-0428-425d-97fc-fa5249c64413.png">

## Release note

We removed support for the `console.proxyFilter` and `console.proxyConfig` Kibana setting in favour of the standard `server.ssl.*` and `elasticsearch.ssl.keyPassphrase` settings.

Fixes https://github.com/elastic/kibana/issues/39494
